src/components/menus/ContextMenu/ContextMenu.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import * as styles from './ContextMenu.scss';
+import { DotsIcon } from '../../icons/Icons';
+import IReactComponentProps from '../../../common/structures/IReactComponentProps';
+import { FunctionGeneric } from '../../../common/structures/Generics';
+import { Tooltip, TooltipProps } from '../../overlays/Tooltip/Tooltip';
+import { PrimaryButton } from '../../buttons/PrimaryButton/PrimaryButton';
+import { TextButton } from '../../buttons/TextButton/TextButton';
+
+export interface IMenuItem {
+    color?: 'red' | 'none';
+    className?: string;
+    content?: React.ReactNode;
+    label?: string;
+    onClick?: FunctionGeneric;
+    enabled?: boolean;
+    type?: 'normal' | 'separator';
+}
+
+export interface IContextMenuProps extends IReactComponentProps {
+    /** className for the list items' container */
+    classNameList?: string;
+    /** className for an individual list item */
+    classNameListItem?: string;
+    /** whether to force the tooltip to show and ignore mouse events **/
+    forceHover?: boolean;
+    /** array of menu items */
+    items: IMenuItem[];
+    popperOptions?: TooltipProps;
+    position?: 'top' | 'bottom';
+    useClickInsteadOfHover?: boolean;
+}
+
+const ContextMenu = (props: IContextMenuProps) => {
+    const {
+        className,
+        classNameList,
+        classNameListItem,
+        items,
+        popperOptions,
+        position,
+        useClickInsteadOfHover,
+        ...otherProps
```

Comment:
Yeah this is pretty common in the repo, eslint-react by default does not like it though since it says it makes it unclear which props are being sent to the component... I'd have to disagree with them there